Support static relocation of /etc/inittab

(c) 2014-2019 by Guenther Brunthaler.
This patch is free software.
Distribution is permitted under the terms of the GPLv2.

Index: busybox/init/init.c
===================================================================
--- busybox.orig/init/init.c
+++ busybox/init/init.c
@@ -33,6 +33,17 @@
 //config:	help
 //config:	Allow init to read an inittab file when the system boot.
 //config:
+//config:config INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B
+//config:	string "Location of inittab file"
+//config:	default "/etc/inittab"
+//config:	depends on FEATURE_USE_INITTAB
+//config:	help
+//config:	This is the location where init actually reads its inittab from.
+//config:
+//config:	Specifying a non-standard location here allows Busybox's init to
+//config:	co-exist with other init implementations using a different
+//config:	inittab at the standard location (/etc/inittab).
+//config:
 //config:config FEATURE_KILL_REMOVED
 //config:	bool "Support killing processes that have been removed from inittab"
 //config:	default n
@@ -670,8 +681,9 @@ static void new_init_action(uint8_t acti
 static void parse_inittab(void)
 {
 #if ENABLE_FEATURE_USE_INITTAB
+# define INITTAB_PATH CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B
 	char *token[4];
-	parser_t *parser = config_open2("/etc/inittab", fopen_for_read);
+	parser_t *parser = config_open2(INITTAB_PATH, fopen_for_read);
 
 	if (parser == NULL)
 #endif
@@ -916,7 +928,7 @@ static void reload_inittab(void)
 {
 	struct init_action *a, **nextp;
 
-	message(L_LOG, "reloading /etc/inittab");
+	message(L_LOG, "reloading " INITTAB_PATH);
 
 	/* Disable old entries */
 	for (a = G.init_action_list; a; a = a->next)
@@ -1225,12 +1237,12 @@ int init_main(int argc UNUSED_PARAM, cha
 //usage:#define init_full_usage "\n\n"
 //usage:       "Init is the first process started during boot. It never exits."
 //usage:	IF_FEATURE_USE_INITTAB(
-//usage:   "\n""It (re)spawns children according to /etc/inittab."
+//usage:   "\n""It (re)spawns children according to " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B "."
 //usage:   "\n""Signals:"
-//usage:   "\n""HUP: reload /etc/inittab"
+//usage:   "\n""HUP: reload " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B ""
 //usage:	)
 //usage:	IF_NOT_FEATURE_USE_INITTAB(
-//usage:   "\n""This version of init doesn't use /etc/inittab,"
+//usage:   "\n""This version of init doesn't use " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B ","
 //usage:   "\n""has fixed set of processed to run."
 //usage:   "\n""Signals:"
 //usage:	)
@@ -1242,7 +1254,7 @@ int init_main(int argc UNUSED_PARAM, cha
 //usage:	"This version of init is designed to be run only by the kernel.\n"
 //usage:	"\n"
 //usage:	"BusyBox init doesn't support multiple runlevels. The runlevels field of\n"
-//usage:	"the /etc/inittab file is completely ignored by BusyBox init. If you want\n"
+//usage:	"the " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B " file is completely ignored by BusyBox init. If you want\n"
 //usage:	"runlevels, use sysvinit.\n"
 //usage:	"\n"
 //usage:	"BusyBox init works just fine without an inittab. If no inittab is found,\n"
@@ -1258,7 +1270,7 @@ int init_main(int argc UNUSED_PARAM, cha
 //usage:	"	tty3::askfirst:/bin/sh\n"
 //usage:	"	tty4::askfirst:/bin/sh\n"
 //usage:	"\n"
-//usage:	"If you choose to use an /etc/inittab file, the inittab entry format is as follows:\n"
+//usage:	"If you choose to use an " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B " file, the inittab entry format is as follows:\n"
 //usage:	"\n"
 //usage:	"	<id>:<runlevels>:<action>:<process>\n"
 //usage:	"\n"
@@ -1315,13 +1327,13 @@ int init_main(int argc UNUSED_PARAM, cha
 //usage:	"\n"
 //usage:	"		Unrecognized actions (like initdefault) will cause init to emit an\n"
 //usage:	"		error message, and then go along with its business. All actions are\n"
-//usage:	"		run in the order they appear in /etc/inittab.\n"
+//usage:	"		run in the order they appear in " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B ".\n"
 //usage:	"\n"
 //usage:	"	<process>:\n"
 //usage:	"\n"
 //usage:	"		Specifies the process to be executed and its command line.\n"
 //usage:	"\n"
-//usage:	"Example /etc/inittab file:\n"
+//usage:	"Example " CONFIG_INITTAB_LOCATION_23J9JCQH4QWSFNCURF9A1L06B " file:\n"
 //usage:	"\n"
 //usage:	"	# This is run first except when booting in single-user mode\n"
 //usage:	"	#\n"
